packages/webdriverio/tests/commands/browser/setWindowSize.test.js

```diff
@@ -0,0 +1,64 @@
+import request from 'request'
+import { remote } from '../../../src'
+
+describe('setWindowSize', () => {
+    let browser
+
+    beforeAll(async () => {
+        browser = await remote({
+            baseUrl: 'http://foobar.com',
+            capabilities: {
+                browserName: 'foobar'
+            }
+        })
+    })
+
+    it('should resize W3C browser window', async () => {
+        await browser.setWindowSize(777, 888)
+        expect(request.mock.calls[1][0].method).toBe('POST')
+        expect(request.mock.calls[1][0].uri.path).toBe('/wd/hub/session/foobar-123/window/rect')
+        expect(request.mock.calls[1][0].body).toEqual({ x: null, y: null, width: 777, height: 888 })
+    })
+
+    it('should resize NO-W3C browser window', async () => {
+        browser = await remote({
+            baseUrl: 'http://foobar.com',
+            capabilities: {
+                browserName: 'foobar-noW3C'
+            }
+        })
+
+        await browser.setWindowSize(999, 1111)
+        expect(request.mock.calls[1][0].method).toBe('POST')
+        expect(request.mock.calls[1][0].uri.path).toBe('/wd/hub/session/foobar-123/window/current/size')
+        expect(request.mock.calls[1][0].body).toEqual({ width: 999, height: 1111 })
+    })
+
+    describe('input checks', () => {
+        it('should throw error if width or height is not number', () => {
+            const invalidTypeError = 'setWindowSize expects width and height of type number'
+
+            expect(() => browser.setWindowSize('777', 888))
```

Comment:
Are you sure these browser calls should not be awaited?

Reply:
@mgrybyk I have tried to await it but it becomes very clumsy , something like `expect((async () => await browser.setWindowSize('777', 888))()).rejects.toEqual(invalidTypeError )` does that look ok? It actually throws an error before it comes to the point we call resizing method so I assumed `await` is not mandatory.

Reply:
@krgolovan can you try to now throw in the command to see if this is being caught?

Reply:
@christian-bromann I have commented out my input checks from `setWindowSize.js`. If parameter type is wrong it throws an error from `command.js line 68` and I can catch it in my test no wait needed. But if parameter is out of range it returns a promise and wait is needed but the problem is: it resolves to `{}` instead of rejecting the promise. I also tried to pass out of range parameter to chrome browser in my boilerplate project and it does throw the error `ERROR webdriver: Request failed due to invalid argument: Parameter 'width' is outside allowed range of [0, 2^31-1].`

Reply:
Try searching for exception handling test examples across repo.
Keywords: toThrow, rejects.

Sorry to say won't be able to help this and next weeks ☹️

Reply:
@mgrybyk When I call `browser._setWindowSize` or `browser.setWindowRect` with a parameter out of range [0, 2^32-1] there is no error or promise rejection so there is nothing to handle in my test. Inside `setWindowSize` function we throw an error for both: invalid type and out of range, and tests are catching all the errors. I just commented out my input checks because @christian-bromann asked me to try catch from command. Initial request was to `await browser.setWindowSize` I can do that if it is the issue.

Reply:
When I call `browser.setWindowSize('777', 888)` in my test it is similar to call something like `function () { throw new Error('input is out of range') }` there is basically nothing to await.